packages/design-system/src/components/Modal/Modal.tsx

```diff
@@ -1,88 +1,121 @@
-import React from 'react';
-import { useDialogState, DialogProps, Dialog as ReakitDialog } from 'reakit';
-import { useTranslation } from 'react-i18next';
+import React, { ReactElement, ReactNode } from 'react';
+import i18n from 'i18next';
+import { Dialog, DialogBackdrop } from 'reakit/Dialog';
+import { IconName } from '@talend/icons';
 
-import { ButtonSecondary, ButtonPrimary } from '../Button';
+import { ButtonDestructive, ButtonPrimary, ButtonSecondary } from '../Button';
 import { Icon } from '../Icon';
+import { StackHorizontal, StackVertical } from '../Stack';
+import { ButtonPrimaryPropsType } from '../Button/variations/ButtonPrimary';
+import { ButtonSecondaryPropsType } from '../Button/variations/ButtonSecondary';
 
-import * as S from './Modal.style';
+import modalStyles from './Modal.scss';
 
-export type ModalProps = DialogProps & {
-	icon?: string;
-	disclosure: React.ComponentPropsWithRef<any>;
-	subtitle?: string;
-	onClose?: () => void;
-	onValidate?: () => void;
+type ModalIcon = IconName | ReactElement;
+
+function ModalIcon(props: { icon: ModalIcon; 'data-test'?: string }): ReactElement {
+	const { icon, ['data-test']: dataTest } = props;
+	return (
+		<div className={modalStyles['modal-icon']} data-test={dataTest}>
+			{typeof icon === 'string' ? <Icon name={icon} /> : icon}
+		</div>
+	);
+}
+
+export type ModalPropsType = {
+	header: {
+		title: ReactNode;
+		description?: string;
+		icon?: ModalIcon;
+	};
+	onClose: Function;
+	primaryAction?: ButtonPrimaryPropsType & { destructive?: boolean };
```

Comment:
This only works for now because ButtonPrimaryPropsType and ButtonDestructivePropTypes have the same type, more or less. It will break as soon as those evolve. 

```suggestion
	primaryAction?: ButtonPrimaryPropsType || { destructive?: boolean & ButtonDestructivePropTypes };
```